lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -128,23 +139,17 @@ namespace Js
         }
         else
         {
-            int foundIndex = -1;
-            bool parentFound = this->otherParents != nullptr && this->otherParents->MapUntil([&](int index, RecyclerWeakReference<ArrayBufferParent>* item)
+            int foundCount = 0;
+            this->otherParents->MapAndRemoveIf([&](RecyclerWeakReference<ArrayBufferParent>* item)->bool
             {
-                if (item->Get() == parent)
+                if(item->Get() == nullptr)
```

Comment:
nullptr means that parent has gone, so we should clean it up the weakRef object and the list